src/lesson-4/src/actions/createTask.ts

```diff
@@ -0,0 +1,79 @@
+import { createTask } from '../server-util';
+import type { Priority, Status, Task } from '../types';
+import { updateTasks } from './updateTasks';
+
+const getFormValidation = (formData: Omit<Task, 'id'>) => {
+  let valid = true;
+
+  const setError = (elementId: string, text: string) => {
+    document.getElementById(elementId)!.textContent = text;
+    valid = false;
+    return;
+  };
+
+  document.querySelectorAll('.error').forEach((e) => (e.textContent = ''));
+
+  if (formData.title.length < 3) {
+    setError('titleError', 'Min. 3 symbols.');
+  }
+
+  if (formData.description.trim().length < 3) {
+    setError('descriptionError', 'Min. 3 symbols.');
+  }
+
+  if (!formData.status) {
+    setError('statusError', 'Select status');
+  }
+
+  if (!formData.priority) {
+    setError('priorityError', 'Select priority');
+  }
+
+  const today = new Date().toISOString().split('T')[0];
+  const deadline = document.querySelector<HTMLInputElement>('#deadline')!;
+  if (!formData.deadline || deadline.value < today) {
+    setError('dateError', 'Data must be in future');
+  }
+
+  return valid;
+};
+
+export const setupForm = () => {
+  const item = document.querySelector<HTMLFormElement>('#simpleForm')!;
```

Comment:
якщо потрібен пошук по id елемента, то краще юзати getElementById, він працює швидше
```suggestion
  const item = document.getElementById<HTMLFormElement>('simpleForm')!;
```